### PR TITLE
Release: add notify-changelog-cut.sh script

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
Adding a script to notify open PRs that the CHANGELOG has just been just. I've just run it.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: tools/release/notify-changelog-cut.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  echo "- ${PR_ID}: ${PR_TITLE}"
+
+  # Check if the notification has already been posted.
+  # We suppress the linter warning about looking for the extract substring instead of doing a regex
+  # match, but that's exactly what we want.
+  # shellcheck disable=SC2076
+  if [[ ${PR_LABELS} =~ "${NOTIFICATION_LABEL}" ]]; then
+    echo "  Notification already posted, nothing to do"
+    continue
+  fi
```
Comment (colega) [inline tools/release/notify-changelog-cut.sh:31]: This won't work for PRs spanning for several releases, but not sure if that's important to cover.

**Comment by colega** [comment]
We can also check if PR has touched the `CHANGELOG.md` before commenting:

```shell
updated=$(gh pr view --json files -t '{{ range .files }}{{ if eq .path "CHANGELOG.md" }}CHANGELOG.md updated{{end}}{{ end }}' "$PR_ID")
if [[ ! -z "$updated" ]]
then 
    // updated, so comment
else
    // not updated, no need to comment
fi
```

**Review by colega — APPROVED** [review]

**Comment by pracucci** [comment]
> We can also check if PR has touched the CHANGELOG.md before commenting:

I thought about it, but I didn't do it intentionally. Reason is that if the CHANGELOG entry is added later, but without rebasing, they will never notice the CHANGELOG was cut.

File: tools/release/notify-changelog-cut.sh (hunk 1 above)
Comment (pracucci) [reply tools/release/notify-changelog-cut.sh:33]: True. Let's see if it will be a real issue in the future :)

Comment (pracucci) [reply tools/release/notify-changelog-cut.sh:33]: Done is better than perfect! 😉

**Pull request merged**